provider/ecs/ecs.go

```diff
@@ -200,103 +202,128 @@ func (p *Provider) loadECSConfig(ctx context.Context, client *awsClient) (*types
 // and the EC2 instance data
 func (p *Provider) listInstances(ctx context.Context, client *awsClient) ([]ecsInstance, error) {
 	var taskArns []*string
-	req, _ := client.ecs.ListTasksRequest(&ecs.ListTasksInput{
-		Cluster:       &p.Cluster,
-		DesiredStatus: aws.String(ecs.DesiredStatusRunning),
-	})
+	var instances []ecsInstance
+	var clustersArn []*string
+	var clusters Clusters
 
-	for ; req != nil; req = req.NextPage() {
-		if err := wrapAws(ctx, req); err != nil {
-			return nil, err
-		}
+	if p.AutoDiscoverClusters {
+		input := &ecs.ListClustersInput{}
+		result, _ := client.ecs.ListClusters(input)
+		for result.NextToken != nil {
+			clustersArn = append(clustersArn, result.ClusterArns...)
 
-		taskArns = append(taskArns, req.Data.(*ecs.ListTasksOutput).TaskArns...)
+			input.NextToken = result.NextToken
+			result, _ = client.ecs.ListClusters(input)
+		}
+		clustersArn = append(clustersArn, result.ClusterArns...)
```

Comment:
I think that result can never be nil. But I like the suggestion, this is simplifying the code. 
I will do the change